src/framework/core/js/ResourceLoader-browser.js

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright The Infusion copyright holders
+See the AUTHORS.md file at the top-level directory of this distribution and at
+https://github.com/fluid-project/infusion/raw/master/AUTHORS.md.
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_3_0_0 = fluid_3_0_0 || {};
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.resourceLoader.UrlClass = URL;
+
+    fluid.resourceLoader.configureXHR = function (xhr, options) {
+        fluid.resourceLoader.loaders.XHR.copyProps.forEach(function (prop) {
+            if (fluid.isValue(options[prop])) {
+                xhr[prop] = options[prop];
+            }
+        });
+        fluid.each(options.requestHeaders, function (value, key) {
+            var values = fluid.makeArray(value);
+            values.forEach(function (oneValue) {
+                xhr.setRequestHeader(key, oneValue);
+            });
+        });
+    };
+
+    fluid.resourceLoader.loaders.XHR = function (resourceSpec) {
+        var togo = fluid.promise();
+        var xhr = resourceSpec.xhr = new XMLHttpRequest();
+        togo.then(null, null, function () {
+            xhr.abort();
+        });
+        var sendSuccess = function () {
+            fluid.invokeLater(function () {
+                var response = !xhr.responseType || xhr.responseType === "text" ? xhr.responseText : xhr.response;
+                togo.resolve(response);
+            });
+        };
+        var sendError = function () {
+            fluid.invokeLater(function () {
+                togo.reject({
+                    isError: true,
+                    status: xhr.status,
+                    textStatus: xhr.statusText
+                });
+            });
+        };
+        xhr.addEventListener("load", function () {
+            var isError = fluid.dataSource.URL.isErrorStatus(xhr.status);
+            isError ? sendError() : sendSuccess();
+        });
+        xhr.addEventListener("error", sendError);
+        var options = $.extend({async: true}, resourceSpec.options);
+        xhr.open(options.method || "GET", resourceSpec.url, options.async, options.username, options.password);
+        fluid.resourceLoader.configureXHR(xhr, options);
+        xhr.send();
+        return togo;
+    };
```

Comment:
Would it be possible/easier to implement this using the [fetch api](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).